src/backend/gl/src/native.rs

```diff
@@ -38,6 +43,79 @@ impl Fence {
     }
 }
 
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub(crate) enum BindingTypes {
```

Comment:
sounds like we could just use `hal::pso::DecriptorType` instead of this enum

Reply:
`hal::pso::DecriptorType` has more types then we need. Once more of the binding types get implemented (I've only tested `SampledImages`) I expect a lot of them to get merged together (ex: `SampledImages` & `SeperateSamplers`).